fw/cache.c

```diff
@@ -2224,32 +2233,43 @@ cache_req_process_node(TfwHttpReq *req, tfw_http_cache_cb_t action)
 		}
 	}
 
-	resp = tfw_cache_build_resp(req, ce, lifetime, id);
-	/*
-	 * The stream of HTTP/2-request should be closed here since we have
-	 * successfully created the resulting response from cache and will
-	 * send this response to the client (without forwarding request to
-	 * the backend), thus the stream will be finished.
-	 */
-	if (resp && TFW_MSG_H2(req)) {
-		id = tfw_h2_stream_id_close(req, HTTP2_HEADERS,
-					    HTTP2_F_END_STREAM);
-		if (unlikely(!id)) {
-			tfw_http_msg_free((TfwHttpMsg *)resp);
-			tfw_http_conn_msg_free((TfwHttpMsg *)req);
-			goto put;
+	T_DBG("Cache: req->method = %d, req->cache_ctl.flags = 0x%x (is not X-Tempesta-Cache: %d)\n",
+		(int)req->method, req->cache_ctl.flags,
+		(int)((req->cache_ctl.flags & TFW_HTTP_CC_CACHE_PURGE) == 0));
+	if (likely(req->method == TFW_HTTP_METH_GET &&
```

Comment:
With this check we break caching for all non-GET requests, which is wrong.

Reply:
done.